cranelift-wasm/src/code_translator.rs

```diff
@@ -132,6 +132,10 @@ pub fn translate_operator<FE: FuncEnvironment + ?Sized>(
             let (arg1, arg2, cond) = state.pop3();
             state.push1(builder.ins().select(cond, arg1, arg2));
         }
+        Operator::TypedSelect { ty: _ } => {
+            let (arg1, arg2, cond) = state.pop3();
+            state.push1(builder.ins().select(cond, arg1, arg2));
```

Comment:
Just to make sure I understand correctly: the reason we don't care about `ty` is because we already validated the wasm (either in wasmtime or spidermonkey) and know that it is a supertype of both `arg1` and `arg2`, and the `builder.ins().select(..)` just treats them all reference types as `r32` or `r64`, right?

Reply:
Yes, `wasmparser` itself contains validation code which will check that `arg1` and `arg2` are subtypes of the specified type. Spidermonkey uses its own equivalent code. And after validation, there's not a need for a stricter type in Cranelift IR, so it all becomes R32/R64.

I'll add a comment to this effect in the code here.